build.cake

```diff
@@ -0,0 +1,408 @@
+            CopyFileToDirectory($"{toolsFolder}/xunit.runner.console/tools/xunit.console.exe", instanceFolder);
+            CopyFileToDirectory($"{toolsFolder}/xunit.runner.console/tools/xunit.runner.utility.desktop.dll", instanceFolder);
+            var logFile = $"{logFolder}/{project}-{framework}-result.xml";
+            var xunitSettings = new XUnit2Settings
+            {
+                ToolPath = $"{instanceFolder}/xunit.console.exe",
+                ArgumentCustomization = builder =>
+                {
+                    builder.Append("-xml");
+                    builder.Append(logFile);
+                    return builder;
+                }
+            };
+            xunitSettings.ExcludeTrait("category", new[] { "failing" });
+            XUnit2($"{instanceFolder}/{project}.dll", xunitSettings);
+        }
+    }
+});
+
+Task("OnlyPublish")
+    .Does(() =>
+{
+    var project = buildPlan.MainProject;
+    foreach (var framework in buildPlan.Frameworks)
+    {
+        foreach (var runtime in buildPlan.Rids)
+        {
+            var outputFolder = $"{publishFolder}/{project}/{runtime}/{framework}";
+            var exitCode = StartProcess(dotnetcli, 
+                new ProcessSettings
+                {
+                    Arguments = $"publish --framework {framework} --runtime {runtime} " +
+                                    $"--configuration {configuration} --output {outputFolder} " +
+                                    $"{sourceFolder}/{project}"
+                });
+            if (exitCode != 0)
+            {
+                throw new Exception($"Failed to publish {project} / {framework}");
+            }
+            
+            // Remove version number on Windows
+            var runtimeShort = Regex.Replace(runtime, "(\\d|\\.)*-", "-");
+            // Simplify Ubuntu to Linux
+            runtimeShort = runtimeShort.Replace("ubuntu", "linux");
+            var buildIdentifier = $"{runtimeShort}-{framework}";
+            // Linux + dnx451 is renamed to Mono
+            if (runtimeShort.Contains("linux-") && framework.Equals("dnx451"))
+                buildIdentifier ="linux-mono";
+            // No need to package OSX + dnx451
+            else if (runtimeShort.Contains("osx-") && framework.Equals("dnx451"))
+                continue;
+            
+            DoArchive(runtime, outputFolder, $"{packageFolder}/{buildPlan.MainProject.ToLower()}-{buildIdentifier}");
+        }
+    }
+});
+
+Task("Publish")
+    .IsDependentOn("Restore")
+    .IsDependentOn("OnlyPublish")
+    .Does(() =>
+{
+});
+
+Task("TestPublish")
+    .IsDependentOn("Publish")
+    .IsDependentOn("RestrictToLocalRuntime")
+    .Does(() =>
+{
+    var project = buildPlan.MainProject;
+    foreach (var framework in buildPlan.Frameworks)
+    {
+        // Skip testing mono executables
+        if (!IsRunningOnWindows() && !framework.Equals("dnxcore50"))
+            continue;
```

Comment:
I notice that this is your coding style. However generally we add curly braces in all conditions. This is the convention here in `OmniSharp`.